ciao-image/datastore/store.go

```diff
@@ -159,20 +159,20 @@ func (s *ImageStore) UploadImage(ID string, body io.Reader) error {
 	img.State = Saving
 
 	if s.rawDs != nil {
-		err := s.rawDs.Write(ID, body)
+		err = s.rawDs.Write(ID, body)
 		if err != nil {
-			return err
+			img.State = Killed
+			err = s.rawDs.Delete(ID)
```

Comment:
I don't think the Delete is needed as Write should clean up if there is an error.  Also, as Write has already cleaned up, we're almost guaranteed that Delete will fail and will overwrite the original error from Write.  I suggest we just get rid of this line.